### PR TITLE
DevOps - Make RPC node accessible over TLS endpoint

### PR DESCRIPTION
**Pull request by ahhda**
Adds TLS support to RPC node using [Caddy](https://caddyserver.com/) and [nip.io](https://nip.io/)

One step closer to #2406

<details>
<summary>Manual Testing</summary>

Screenshot 2021-06-02 at 11 21 42 PM

Screenshot 2021-06-02 at 11 21 46 PM

</details>

### CODE REVIEW

File: devops/infrastructure/roles/rpc/tasks/main.yml [hunk 1]
```diff
@@ -20,3 +20,17 @@
...
+    apply:
+      become: yes
+  vars:
+    caddy_config: "{{ lookup('template', 'templates/Caddyfile.j2') }}"
+    caddy_systemd_capabilities_enabled: true
+    public_ip: "{{ inventory_hostname }}"
+
+- name: Print RPC node DNS
+  debug:
+    msg: "RPC Node DNS is: https://{{ inventory_hostname }}.nip.io"
```
Comment (mnaamani) [inline devops/infrastructure/roles/rpc/tasks/main.yml]:
```suggestion
    msg: "RPC Endpoint: wss://{{ inventory_hostname }}.nip.io"
```

File: devops/infrastructure/roles/rpc/templates/Caddyfile.j2 [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{{ public_ip }}.nip.io
+
+reverse_proxy localhost:9944
```
Comment (mnaamani) [inline devops/infrastructure/roles/rpc/templates/Caddyfile.j2]: Its a a good idea to also proxy the http-rpc port 9933. So maybe have two pathes one for websocket and other for the non websocket endpoint:

{{ public_ip }}.nip.io/ws-rpc  --> reverse_proxy localhost:9944
and
{{ public_ip }}.nip.io/http-rpc -> reverse_proxy localhost:9933

and have these paths be also based in as template variables from `roles/rpc/tasks/main.yml`

**Review by mnaamani — CHANGES_REQUESTED** [review]
Works great. I just left request to also expose the http-rpc port

File: devops/infrastructure/roles/rpc/templates/Caddyfile.j2 (hunk 2 above)
Comment (ahhda) [reply devops/infrastructure/roles/rpc/templates/Caddyfile.j2]: Done

File: devops/infrastructure/roles/rpc/tasks/main.yml (hunk 1 above)
Comment (ahhda) [reply devops/infrastructure/roles/rpc/tasks/main.yml]: Done

**Review by mnaamani — APPROVED** [review]

**Pull request merged**